src/Compilers/CSharp/Test/Semantic/Semantics/RefExtensionMethodsTests.cs

```diff
@@ -0,0 +1,1151 @@
+                // (20,11): error CS1061: 'int' does not contain a definition for 'Print' and no extension method 'Print' accepting a first argument of type 'int' could be found (are you missing a using directive or an assembly reference?)
+                //         0.Print();                  // Error
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "Print").WithArguments("int", "Print").WithLocation(20, 11));
+        }
+
+        [Fact]
+        public void AmbiguousRefnessForExtensionMethods_LValue()
+        {
+            var code = @"
+public static class Ext1
+{
+    public static void Print(ref int p)
+    {
+        System.Console.WriteLine(p);
+    }
+}
+public static class Ext2
+{
+    public static void Print(ref readonly int p)
+    {
+        System.Console.WriteLine(p);
+    }
+}
+public class Program
+{
+    public static void Main()
+    {
+        int value = 0;
+        value.Print();              // Error
+        
+        Ext1.Print(ref value);      // Ok
+        Ext2.Print(value);          // Ok
+    }
+}";
+
+            CreateCompilationWithMscorlibAndSystemCore(code).VerifyDiagnostics(
+                // (21,15): error CS1061: 'int' does not contain a definition for 'Print' and no extension method 'Print' accepting a first argument of type 'int' could be found (are you missing a using directive or an assembly reference?)
+                //         value.Print();              // Error
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "Print").WithArguments("int", "Print").WithLocation(21, 15));
+        }
+
+        [Fact]
+        public void ReadOnlynessPreservedThroughMultipleCalls()
+        {
+            var code = @"
+public static class Ext
+{
+    public static void ReadOnly(ref readonly int p)
+    {
+        Ref(ref p);     // Should be an error
+    }
+    public static void Ref(ref int p)
+    {
+        Ref(ref p);     // Should be OK
+    }
+}";
+
+            CreateCompilationWithMscorlibAndSystemCore(code).VerifyDiagnostics(
+                // (6,17): error CS8406: Cannot use variable 'in int' as a ref or out value because it is a readonly variable
+                //         Ref(ref p);     // Should be an error
+                Diagnostic(ErrorCode.ERR_RefReadonlyNotField, "p").WithArguments("variable", "in int").WithLocation(6, 17));
+        }
+
+        [Fact]
+        public void SemanticModelPreservesRefness_RefExtensionMethods()
+        {
+            var code = @"
+public static class Ext
+{
+    public static void Method(ref int p) { }
+}";
+
+            var comp = CreateCompilationWithMscorlibAndSystemCore(code).VerifyDiagnostics();
+            var tree = comp.SyntaxTrees.Single();
+            var parameter = tree.FindNodeOrTokenByKind(SyntaxKind.Parameter);
+            Assert.True(parameter.IsNode);
+
+            var model = comp.GetSemanticModel(tree);
+            var symbol = (ParameterSymbol)model.GetDeclaredSymbolForNode(parameter.AsNode());
```

Comment:
>var symbol = (ParameterSymbol)model.GetDeclaredSymbolForNode(parameter.AsNode()); [](start = 12, length = 81)

It is an overkill to use semantic model to find non-lambda/non-local function parameters. We have a helper ```public static T GetMember<T>(this Compilation compilation, string qualifiedName)``` that allows you to get to any member very easily.